syntax = "proto3";

message TaskRequest {
  string task_id = 1;
  string script = 2;
  bytes input = 3;   // any serial
}

message TaskResult {
  string task_id = 1;
  bool success = 2;        
  bytes output = 3;  
  string error_message = 4; 
}

message WorkerRegister {
    string worker_id = 1; 
}
